workflows/csharp/sdk/order-processor/Program.cs

```diff
@@ -50,31 +47,28 @@
 // Start the workflow
 Console.WriteLine("Starting workflow {0} purchasing {1} {2}", orderId, ammountToPurchase, itemToPurchase);
 
-await workflowClient.ScheduleNewWorkflowAsync(
-    name: nameof(OrderProcessingWorkflow),
-    instanceId: orderId,
-    input: orderInfo);
-
-// Wait a second to allow workflow to start
-await Task.Delay(TimeSpan.FromSeconds(1));
+await daprClient.StartWorkflowAsync(
+    workflowComponent: DaprWorkflowComponent,
+    workflowName: nameof(OrderProcessingWorkflow),
+    input: orderInfo,
+    instanceId: orderId);
 
-WorkflowState state = await workflowClient.GetWorkflowStateAsync(
+// Wait for the workflow to start and confirm the input
+GetWorkflowResponse state = await daprClient.WaitForWorkflowStartAsync(
     instanceId: orderId,
-    getInputsAndOutputs: true);
+    workflowComponent: DaprWorkflowComponent);
 
 Console.WriteLine("Your workflow has started. Here is the status of the workflow: {0}", state.RuntimeStatus);
-while (!state.IsWorkflowCompleted)
-{
-    await Task.Delay(TimeSpan.FromSeconds(5));
-    state = await workflowClient.GetWorkflowStateAsync(
-        instanceId: orderId,
-        getInputsAndOutputs: true);
-}
+
+// Wait for the workflow to complete
+state = await daprClient.WaitForWorkflowCompletionAsync(
+    instanceId: orderId,
+    workflowComponent: DaprWorkflowComponent);
 
 Console.WriteLine("Workflow Status: {0}", state.RuntimeStatus);
 
 void RestockInventory()
 {
-    daprClient.SaveStateAsync<OrderPayload>(storeName, "Cars",  new OrderPayload(Name: "Cars", TotalCost: 15000, Quantity: 100));
+    daprClient.SaveStateAsync<OrderPayload>(StoreName, "Cars",  new OrderPayload(Name: "Cars", TotalCost: 15000, Quantity: 100));
```

Comment:
Can we change "Cars" to itemToPurchase?

Reply:
agreed should use variables for itemToPurchase and ammountToPurchase